azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java

```diff
@@ -40,23 +41,25 @@
 @Mojo(name = "deploy", defaultPhase = LifecyclePhase.DEPLOY)
 public class DeployMojo extends AbstractWebAppMojo {
     private static final Path FTP_ROOT = Paths.get("/site/wwwroot");
-    private static final String NO_RESOURCES_CONFIG = "<resources> is empty. Please make sure it is configured in pom.xml.";
-
-    public static final String WEBAPP_NOT_EXIST = "Target Web App doesn't exist. Creating a new one...";
-    public static final String WEBAPP_CREATED = "Successfully created Web App.";
-    public static final String CREATE_DEPLOYMENT_SLOT = "Target Deployment Slot doesn't exist. Creating a new one...";
-    public static final String CREATE_DEPLOYMENT_SLOT_DONE = "Successfully created the Deployment Slot.";
-    public static final String UPDATE_WEBAPP = "Updating target Web App...";
-    public static final String UPDATE_WEBAPP_SKIP = "No runtime configured. Skip the update.";
-    public static final String UPDATE_WEBAPP_DONE = "Successfully updated Web App.";
-
-    public static final String WEBAPP_NOT_EXIST_FOR_SLOT = "The Web App specified in pom.xml does not exist. " +
+    private static final String CREATE_WEBAPP = "Creating web app...";
+    private static final String CREATE_WEB_APP_DONE = "Successfully created Web App.";
+    private static final String UPDATE_WEBAPP = "Updating target Web App...";
+    private static final String UPDATE_WEBAPP_DONE = "Successfully updated Web App.";
+    private static final String CREATE_RESOURCE_GROUP = "Creating resource group %s in region %s...";
+    private static final String CREATE_RESOURCE_GROUP_DONE = "Successfully created resource group %s.";
+    private static final String CREATE_APP_SERVICE_PLAN = "Creating app service plan...";
+    private static final String CREATE_APP_SERVICE_DONE = "Successfully created app service plan %s.";
+    private static final String WEBAPP_NOT_EXIST_FOR_SLOT = "The Web App specified in pom.xml does not exist. " +
             "Please make sure the Web App name is correct.";
-    public static final String SLOT_SHOULD_EXIST_NOW = "Target deployment slot still does not exist. " +
-            "Please check if any error message during creation.";
+    private static final String CREATE_DEPLOYMENT_SLOT = "Creating deployment slot %s in web app %s";
+    private static final String CREATE_DEPLOYMENT_SLOT_DONE = "Successfully created the Deployment Slot.";
+    private static final String APP_SERVICE_PLAN_NOT_FOUND = "App service plan %s was not found in resource group %s";
 
     @Override
     protected void doExecute() throws AzureExecutionException {
+        // initialize library client
+        az = getOrCreateLibraryClient();
```

Comment:
getOrCreateAzureClient

Reply:
changed to `getOrCreateAzureAppServiceClient`